AGS_Engine/Compute/Extract/ExtractAll.cs

```diff
@@ -0,0 +1,114 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Adapters.AGS;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using FuzzySharp;
+using FuzzySharp.Extractor;
+using FuzzySharp.SimilarityRatio;
+using FuzzySharp.SimilarityRatio.Scorer;
+using FuzzySharp.SimilarityRatio.Scorer.StrategySensitive;
+
+namespace BH.Engine.Adapters.AGS
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Extracts all values when comparing the query to the choices. The method uses the weighted ratio and full process.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("choices", "A list of strings to compare the query against.")]
+        [Input("cutOff", "The cuttoff score (i.e. lower bound) for results to be returned.")]
+        [Output("result", "A FuzzyStringResult containing the strings, scores and indexes resulting from the fuzzy matching algorithm.")]
+        public static FuzzyStringResult ExtractAll(string query, IEnumerable<string> choices, int cutOff = 0)
+        {
+            return ExtractAll(query, choices, Scorer.DefaultRatioScorer, cutOff);
+        }
+
+        /***************************************************/
+
+        [Description("Extracts the top n values with the highest score when comparing the query to the choices. If the default value on n = 0 is used, all values will be returned unsorted.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("choices", "A list of strings to compare the query against.")]
+        [Input("scorer", "The method to use to score the strings when compared.")]
+        [Input("cutOff", "The cuttoff score (i.e. lower bound) for results to be returned.")]
+        [Output("result", "A FuzzyStringResult containing the strings, scores and indexes resulting from the fuzzy matching algorithm.")]
+        public static FuzzyStringResult ExtractAll(string query, IEnumerable<string> choices, Scorer scorer = Scorer.DefaultRatioScorer, int cutOff = 0)
```

Comment:
If you move `cutOff` before `scorer`, you can get rid of the first method entirely. Makes it a bit cleaner in the choice of components.